src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -972,8 +973,56 @@ bool ParserStringLiteral::parseImpl(Pos & pos, ASTPtr & node, Expected & expecte
 
     String s;
 
+    auto makeStringLiteral = [&](String str) -> bool
```

Comment:
I'd suggest making `makeStringLiteral` and `makeHexOrBinStringLiteral` static functions in this file, it may be a bit easier to read without capturing.